From 497513186ad4fbdcbc4c524326f174e312046b11 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 07 Jun 2019 22:30:23 +0300
Subject: [PATCH] [SW-4185] gaudi: pll: raise hbm lbw clock

The PSOC HBM PLL LBW clock can use the regular W/A and is configured
accordingly.
The NIC2 HBM PLL LBW clock cannot use the gated switch as a W/A,
because we will loose the ability to write to its configuration.
Therefore we apply the same W/A as for the NIC2 HBM HBW clock which
includes the same reset procedure.
We configure the NIC2 HBM LBW at the same time as the HBW and use call
the reset afterwards, so only one reset will be needed. Reset in turn
removes the clock switch glitch.

Change-Id: Iafb8125a2980c64d5a035f0ec38dcfc394c10719
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 6900bd5..c09eff5 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -176,12 +176,12 @@
 		.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
+			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
 		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_INIT,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
@@ -218,11 +218,11 @@
 		.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
+			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
+		.div_wa = { HL_PLL_WA_NOWA,	/* Can't gate NIC2_HBM */
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
